monkey/infection_monkey/exploit/weblogic.py

```diff
@@ -66,18 +69,41 @@ def exploit(self, url, command):
             print(e)
         return True
 
-    def check_if_exploitable(self, url):
+    def add_vulnerable_urls(self, urls):
+        """
+        Overrides parent method to use listener server
+        """
         # Server might get response faster than it starts listening to it, we need a lock
         httpd, lock = self._start_http_server()
+        exploitable = False
+
+        for url in urls:
+            if self.check_if_exploitable(url, httpd):
+                exploitable = True
+                break
+
+        if not exploitable and httpd.get_requests < 1:
+            # Wait for responses
+            time.sleep(REQUEST_TIMEOUT)
+
+        if httpd.get_requests > 0:
```

Comment:
Can we avoid having this dup check here and in `check_if_exploitable`?

Reply:
I don't think so, because httpd.get_requests value might change when interpreter gets to this part.